tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,370 @@
+
+    SDK Internals repository - Place we store our markdown files
+    git clone https://github.com/Azure/iot-sdks-internals.git
+
+.EXAMPLE
+.\diffapi
+
+Executes the commands assuming everything is cloned relative to the SDK repository and the AsmDiff tool is compiled using the arcade instructions.
+.EXAMPLE
+.\diffapi -AsmToolExecutable c:\tools\asmdifftool
+
+Specifies the location of the AsmDiff tool executable to run the diff commands.
+.EXAMPLE
+.\diffapi -SDKInternalsPath c:\repo\iot-sdks-internals
+
+Specifies the location of the iot-sdk-internals repository if not cloned relative to the azure-iot-sdk-csharp repository.
+.EXAMPLE
+.\diffapi -IsPreview
+
+Executes the commands assuming you have cloned the iot-sdk-internals repository to a directory relative to the azure-iot-sdk-csharp repository, and if you've installed the asmdiff tool.
+.LINK
+https://github.com/azure/azure-iot-sdk-csharp
+#>
+
+Param(
+    [ValidateScript( {
+            if (-Not ($_ | Test-Path -PathType Leaf)) 
+            {
+                throw "File $_ does not exist."
+            } 
+            elseif ((Get-Command $_).Extension.ToLower() -ne '.exe') 
+            {
+                throw "File $_ is not an executable."
+            }
+            return $true
+        })]
+    # The executable path to the compiled AsmDiff tool found in the dotnet arcade (ex: c:\tools\asmdifftool\dotnet-asmdiff.exe)
+    [System.IO.FileInfo] $AsmToolExecutable = $null,
+    [ValidateScript( {
+            if (-Not ($_ | Test-Path -PathType Container)) 
+            {
+                throw "Folder $_ does not exist."
+            }
+            return $true
+        })]
+    # The path of the iot-sdk-internals repository (ex: c:\repo\iot-sdks-internals)
+    [System.IO.FileInfo] $SDKInternalsPath = $null,
+    # Indicates you will compare the output to the last preview version instead of master
+    [switch] $IsPreview,
+    # Enable debug logging
+    [switch] $Debug
+)
+
+# Enable the debug logging to console
+if ($Debug) 
+{
+    $DebugPreference = "Continue"
+}
+
+# Defaults for both the current repository and the repository parent directory
+$repoRootPath = (Get-Item $pwd).Parent.Parent.FullName
+$baseRootPath = (Get-Item $repoRootPath).Parent.FullName
+
+Write-Debug "Repository root path: $repoRootPath"
+Write-Debug "Repository base path: $baseRootPath"
+
+# Release log file names
+$releaseLogDetailed = "releaselog_detailed.txt"
+$releaseLogShort = "releaselog_short.txt"
+
+# First check to see if we've followed the guide
+$asmToolExecutableCommand = Get-Command dotnet-asmdiff -ErrorAction SilentlyContinue
+if ($null -eq $asmToolExecutableCommand)
+{
+    Write-Debug "Unable to locate dotnet-asmdiff on the command line." 
+}
+
+if ($AsmToolExecutable -ne $null) 
+{
+    Write-Debug "Using user supplied Asm Diff tool executable." 
+    $asmToolExecutableCommand = $AsmToolExecutable
+}
+Write-Debug "AsmDiff executable: $asmToolExecutableCommand"
```

Comment:
what if both $asmToolExecutableCommand and $AsmToolExecutable return null?

Reply:
There is a block below that that has handles this case, I'll move it up to make it more apparent.